noir/scripts/sync.sh

```diff
@@ -181,7 +159,7 @@ function needs_switch {
   # If we're on the wanted checkout, it could be that we rebased on origin in order
   # to push the branch without the fixup commit. In that case we'd need to do a fresh
   # checkout to re-apply the fixup and the patches.
-  if [ $have == true ] && has_fixup_and_patch; then
+  if [ $have == true ] && has_patch_commit; then
```

Comment:
We can probably avoid the empty patch commit by just searching for the commit message in the last patch file.

Reply:
We don't necessarily have to have a patch file, right? I think the empty commit is easy to understand, and easy to see that it finished doing it without having to look into the patches directory when you're just looking at the git log.

Reply:
Another reason is that it makes it easy to see which commit is the first one that I added _on top of_ the local patches, when I want to rebase my feature branch.